to.etc.domui/src/main/java/to/etc/domui/component/wizard/WizardPopupWindow.java

```diff
@@ -0,0 +1,193 @@
+	}
+
+	@Override
+	public void createContent() throws Exception {
+		refreshNavbarContainer();
+		refreshBodyContainer();
+	}
+
+	/**
+	 * This refreshes the div that holds the wizard's navbar.
+	 */
+	public void refreshNavbarContainer() {
+		Div navbar = buildNavbar();
+		Div stepsContainer = buildStepsContainer();
+		navbar.add(stepsContainer);
+		add(navbar);
+	}
+
+	/**
+	 * This refreshes the div that holds the body of the current step.
+	 * @throws Exception
+	 */
+	public void refreshBodyContainer() throws Exception {
+		if(null == m_currentStepBody) {
+			m_currentStepBody = new Div();
+		}
+		m_currentStepBody.removeAllChildren();
+		m_currentStepBody.add(new WizardPopupStepBodyFragment(m_currentStep, getActions(), isFirstStep(), isLastStep()));
+		add(m_currentStepBody);
+	}
+
+	/**
+	 * Returns whether the current step is the first one.
+	 * @return
+	 */
+	private boolean isFirstStep() {
+		return m_steps.indexOf(m_currentStep) == 0;
+	}
+
+	/**
+	 * Returns whether the current step is the last one.
+	 * @return
+	 */
+	private boolean isLastStep() {
+		return m_steps.indexOf(m_currentStep) + 1 == m_steps.size();
+	}
+
+	/**
+	 * Build the wizard's navbar that holds all available steps.
+	 * @return
+	 */
+	private Div buildNavbar() {
+		Div navbar = new Div();
+		navbar.setCssClass("wizard-navbar");
+		return navbar;
+	}
+
+	/**
+	 * This creates a container that holds all available wizard steps
+	 * in the navbar.
+	 * @return
+	 */
+	private Div buildStepsContainer() {
+		Div phasesContainer = new Div();
+		phasesContainer.setCssClass("steps-list-container");
+
+		Iterator<WizardPopupStep> itr = m_steps.iterator();
+		while(itr.hasNext()) {
+			WizardPopupStep phase = itr.next();
+			if(phase.equals(m_currentStep)) {
+				phasesContainer.add(new WizardPopupStepNavbarFragment(phase, true));
+			} else {
+				phasesContainer.add(new WizardPopupStepNavbarFragment(phase, false));
+			}
+			if(itr.hasNext()) {
+				phasesContainer.add(">");
+			}
+		}
+		return phasesContainer;
+	}
+
+	/**
+	 * Create list of available wizard actions (e.g. next button, end button).
+	 * @return
+	 */
+	private Map<WizardPopupStepActions, IUIAction<Void>> getActions() {
+		Map<WizardPopupStepActions, IUIAction<Void>> actions = new HashMap<>();
+		for(WizardPopupStepActions action : WizardPopupStepActions.values()) {
+			actions.put(action, action.getAction(this));
+		}
+		return actions;
+	}
+
+	/**
+	 * With this, one can create wizard step labels. Only the step title label is mandatory.
+	 * If other labels are not provided, the default labels will be used.
+	 * @param titleLabel
+	 * @param cancelButtonLabel
+	 * @param backButtonLabel
+	 * @param nextButtonLabel
+	 * @param endButtonLabel
+	 * @return
+	 */
+	public static Map<String, String> createWizardStepLabels(@Nonnull String titleLabel, @Nullable String cancelButtonLabel, @Nullable String backButtonLabel, @Nullable String nextButtonLabel, @Nullable String endButtonLabel) {
```

Comment:
This seems to repeat. You should definitely not need a map, because what labels to use is very determinate; having fields is way better.